starter-code/routes/authRoutes.js

```diff
@@ -0,0 +1,144 @@
+// routes/auth-routes.js
+const express = require("express");
+const authRoutes = express.Router();
+const passport = require("passport");
+const ensureLogin = require("connect-ensure-login");
+
+// User model
+const User = require("../models/user");
+
+// Bcrypt to encrypt passwords
+const bcrypt = require("bcrypt");
+const bcryptSalt = 10;
+
+authRoutes.get("/login", (req, res, next) => {
+  res.render("auth/login");
+});
+
+authRoutes.post(
+  "/login",
+  passport.authenticate("local", {
+    successRedirect: "admin/userList",
+    failureRedirect: "auth/login",
+    failureFlash: true,
+    passReqToCallback: true
+  })
+);
+
+authRoutes.get("/admin/userList", ensureLogin.ensureLoggedIn(), (req, res) => {
```

Comment:
Notice that you are checking that the user in logged in, but you are not checking whar role he has, so as a guest I had access to this page. The idea is to create a middleware to check whether you are a guest, TA etc.